lib/ansible/modules/cloud/amazon/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,223 @@
+def ses_verify_dkim_domain(module, client, identity):
+    domain = identity.split('@')[-1]
+    try:
+        if not module.check_mode:
+            response = client.verify_domain_dkim(Domain=domain)
+            return response['DkimTokens']
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to start DKIM verification for {domain}.'.format(domain=domain))
+
+
+def set_identity_dkim_enabled(module, client, identity, enabled, retries=10, retry_delay=10):
+    if enabled:
+        operation = 'enable'
+    else:
+        operation = 'disable'
+    for retry in range(1, retries + 1):
+        try:
+            if not module.check_mode:
+                return client.set_identity_dkim_enabled(DkimEnabled=enabled, Identity=identity)
+        except (BotoCoreError) as e:
+            module.fail_json_aws(e, msg='Failed to {operation} DKIM for {identity}.'.format(
+                operation=operation,
+                identity=identity,
+            ))
+        except (ClientError) as e:
+            error = e.response.get('Error', {})
+            error_code = error.get('Code', 'Unknown')
+            error_message = error.get('Message', 'Unknown')
+            # verify_dkim_domain seems to take some time to replicate the status consistently to
+            # other parts of the SES API. So if we get this specific client error we retry after
+            # a delay  so that we correctly enable or disable DKIM.
+            # Note we need this even when disabling since this seems to fail tests sometimes even
+            # after having succeeded once. It seems that we get inconsistent failures for some time
+            # until the domain verification attempt is fully propagated.
+            if (retry >= retries or
+                    not (error_code == 'InvalidParameterValue' and 'not verified for DKIM signing' in error_message)):
+                module.fail_json_aws(e, msg='Failed to {operation} DKIM for {identity}.'.format(
+                    operation=operation,
+                    identity=identity,
+                ))
+            time.sleep(retry_delay)
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            "identity": dict(required=True, type='str'),
+            "state": dict(default='enabled', choices=['enabled', 'disabled']),
+        },
+        supports_check_mode=True,
+    )
+
+    # SES APIs seem to have a much lower throttling threshold than most of the rest of the AWS APIs.
+    # Docs say 1 call per second. This shouldn't actually be a big problem for normal usage, but
+    # the ansible build runs multiple instances of the test in parallel that's caused throttling
+    # failures so apply a jittered backoff to call SES calls.
+    client = module.client('ses', retry_decorator=AWSRetry.jittered_backoff())
+
+    identity = module.params.get('identity')
+    state = module.params.get('state')
+    changed = False
+
+    state_check = _get_state_check(state)
```

Comment:
So, I guess we have to fix it here to allow 'any' state while in `check_mode`. Simillar as in line 216.

Reply:
We shouldn't need to (but perhaps I'm missing something here). The state_check function here is used for two purposes:

1. Checking whether the identity is already in the right state to decide if we need to make any changes.
2. Waiting until the get_dkim_attributes call returns the right state if we do make any changes

The first should look for the actual state regardless of whether we're in check mode or not because we want to indicate whether or not we would change something.

The second needs to not wait in check mode because no change was made so it would wait for nothing.

Now that I think about it though, there's actually no need to call get_dkim_attributes again on line 218 in check_mode since we didn't actually make any change, so we could replace that wth something like what you've done on line 120ish.